R/module_teal.R

```diff
@@ -184,7 +189,32 @@ srv_teal <- function(id, data, modules, filter = teal_slices()) {
       }
     )
 
-    data_rv <- srv_init_data("data", data = data, modules = modules, filter = filter)
+    init_data <- srv_init_data("data", data = data, modules = modules, filter = filter)
+    data_rv <- reactive({
+      if (inherits(init_data(), "qenv.error")) {
+        teal_data()
+      } else {
+        init_data()
+      }
+    })
+
+    output$shiny_error <- renderUI({
+      if (inherits(init_data(), "qenv.error")) {
+        validate(
+          need(
+            FALSE,
+            paste(
+              "Error when executing the `data` module:",
+              strip_style(paste(init_data()$message, collapse = "\n")),
+              "\nCheck your inputs or contact app developer if error persists.",
+              collapse = "\n"
+            )
+          )
+        )
+      }
```

Comment:
same code is used in `srv_validate_reactive_teal_data`. As mentioned [here](https://github.com/insightsengineering/teal/pull/1330#discussion_r1734330124) please find common ground in those places where validation occur and encapsulate in a module.